### PR TITLE
Implement FPlasticSourceControlMenu::OpenDeskoptApp()

### PR DESCRIPTION
**Pull request by SRombautsU**
- Checks the Windows Registry to find Unity Version Control install location
- Implemented and test for both the regular "Plastic" GUI and Gluon

Also implemented the correct application & path for MacOS and symlinks for Linux, but they are not supported and not tested at all.

### CODE REVIEW

File: Source/PlasticSourceControl/Private/PlasticSourceControlMenu.cpp [hunk 1]
```diff
@@ -31,6 +33,11 @@
 #include "ToolMenus.h"
 #include "ToolMenuMisc.h"
 
+#if PLATFORM_WINDOWS
+#include "Windows/AllowWindowsPlatformTypes.h"
+#include "Windows/WindowsPlatformMisc.h"
+#endif
```
Comment (juliomaqueda) [inline Source/PlasticSourceControl/Private/PlasticSourceControlMenu.cpp:39]: In my local environment, these Windows<sup>TM</sup> imports make the native method `GetUserName` re-enter the game, provoking multiple compilation errors.
```suggestion
#if PLATFORM_WINDOWS
#include "Windows/AllowWindowsPlatformTypes.h"
#include "Windows/WindowsPlatformMisc.h"
#undef GetUserName
#endif
```

File: Source/PlasticSourceControl/Private/PlasticSourceControlMenu.cpp [hunk 2]
```diff
@@ -529,6 +536,41 @@ void FPlasticSourceControlMenu::VisitLockRulesURLClicked(const FString InOrganiz
...
+	const FString DeskoptAppPath = FPaths::Combine(InstallLocation, FPlasticSourceControlModule::Get().GetProvider().IsPartialWorkspace() ? GluonExe : PlasticExe);
+#elif PLATFORM_MAC
+	const TCHAR* PlasticExe	= "/Applications/PlasticSCM.app/Contents/MacOS/macplasticx";
+	const TCHAR* GluonExe	= "/Applications/Gluon.app/Contents/MacOS/macgluonx";
+	const TCHAR* DeskoptAppPath = FPlasticSourceControlModule::Get().GetProvider().IsPartialWorkspace() ? GluonExe : PlasticExe);
+#elif PLATFORM_LINUX
+	const TCHAR* PlasticExe	= "/usr/bin/plasticgui ";
+	const TCHAR* GluonExe	= "/usr/bin/gluon";
+	const TCHAR* DeskoptAppPath = FPlasticSourceControlModule::Get().GetProvider().IsPartialWorkspace() ? GluonExe : PlasticExe);
+#endif
```
Comment (juliomaqueda) [inline Source/PlasticSourceControl/Private/PlasticSourceControlMenu.cpp:560]: For better readability, I'd recommend to extract this well-packed logic to a separate method or utility class, returning the current `DeskoptAppPath`.

**Review by juliomaqueda — CHANGES_REQUESTED** [review]
Great work @SRombautsU!
Please, checkout the minor comments I added.

File: Source/PlasticSourceControl/Private/PlasticSourceControlMenu.cpp (hunk 1 above)
Comment (SRombautsU) [reply Source/PlasticSourceControl/Private/PlasticSourceControlMenu.cpp]: Fixed, thanks!

File: Source/PlasticSourceControl/Private/PlasticSourceControlMenu.cpp (hunk 2 above)
Comment (SRombautsU) [reply Source/PlasticSourceControl/Private/PlasticSourceControlMenu.cpp]: I moved this OS dependant logic into a new PlasticSourceControlUtils::FindDesktopApplicationPath()

**Pull request merged**